File: python3/FindFirstandLastPositionofElementinSortedArray/34.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l = self.loop(nums,target, True)
        
        if l == len(nums) or nums[l] != target:
            return [-1, -1]
    
        r = self.loop(nums, target, False)
        
        return [l, r - 1]
        
    def loop(self, nums: List[int], target: int, left: bool) -> int:
        l = 0
        r = len(nums)
        
        while l < r:
            mid = (l + r) // 2
            
            if nums[mid] > target or (left and target == nums[mid]):
                r = mid
            else:
                l = mid + 1
                
        return l

File: go/src/FindFirstandLastPositionofElementinSortedArray/34.go
package FindFirstandLastPositionofElementinSortedArray

func searchRange(nums []int, target int) []int {

	l := loop(nums, target, true)

	if l == len(nums) || nums[l] != target {
		return []int{-1, -1}
	}

	r := loop(nums, target, false)

	return []int{l, r - 1}
}

func loop(nums []int, target int, left bool) int {
	l := 0
	r := len(nums)

	for l < r {
		mid := (l + r) / 2

		if nums[mid] > target || (left && target == nums[mid]) {
			r = mid
		} else {
			l = mid + 1
		}
	}

	return l
}
